scripts/dns-zone-and-server-certificates/create-or-renew-one-certificate.sh

```diff
@@ -0,0 +1,131 @@
+
+certificateDescription=$( aws acm --profile $profile --region $certificateRegion describe-certificate \
+                                  --certificate-arn $certificateArn )
+
+echo "Certificate Description"
+echo "$certificateDescription"
+
+certificateStatus=$( echo "$certificateDescription" | jq '.Certificate.Status' | tr -d '"' )
+
+echo "Certificate status = $certificateStatus"
+
+
+if ( [ "PENDING_VALIDATION" = "$certificateStatus" ] ) then
+  echo "Certificate needs validation"
+  validationDnsName=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Name' | tr -d '"' )
+  validationDnsType=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Type' | tr -d '"' )
+  validationDnsValue=$( echo "$certificateDescription" | jq '.Certificate.DomainValidationOptions[0].ResourceRecord.Value' | tr -d '"' )
+
+  echo "validation needs DNS entry"
+  echo " - Name: $validationDnsName"
+  echo " - Type: $validationDnsType"
+  echo " - Value: $validationDnsValue"
+
+  echo "{ \"Changes\": [{ \n"\
+       "    \"Action\": \"CREATE\", \n"\
+       "    \"ResourceRecordSet\": { \n"\
+       "        \"Name\": \"${validationDnsName}\",\n"\
+       "        \"Type\": \"${validationDnsType}\",\n"\
+       "        \"TTL\": 300,"\
+       "        \"ResourceRecords\": [{ \"Value\": \"${validationDnsValue}\"}] \n"\
+       "    }\n"\
+       "}]}" \
+        > /tmp/route53RecordSetChanges.json
+
+  echo "DNS Record change request"
+  cat /tmp/route53RecordSetChanges.json | jq
+
+  echo "Look for hosted zone id"
+  hostedZoneId=$( aws --profile $profile --region $zoneRegion route53 list-hosted-zones-by-name \
+             --dns-name ${zoneName} | jq ".HostedZones[] | select(.Name==\"${zoneName}.\") | .Id"\
+              | sed -e 's|/hostedzone/||' | tr -d '"')
+  if ( [ "" = "$hostedZoneId" ] ) then
+    echo "Hosted zone not found"
+    if ( [ "$BASH_SOURCE" = "" ] ) then
+      return 1
+    else
+      exit 1
+    fi
+  fi
+  echo "hosted zone Id: $hostedZoneId"
+
+  echo "Create DNS records for validation"
+  aws --profile $profile --region $zoneRegion route53 change-resource-record-sets \
+      --hosted-zone-id "${hostedZoneId}" --change-batch file:///tmp/route53RecordSetChanges.json
+
+  echo "Wait for Automatic Validation"
+  certificateStatus=$( echo "$certificateDescription" | jq '.Certificate.Status' | tr -d '"' )
+  counter=1
+  while ( [  "$certificateStatus" != "ISSUED" -a $counter -lt 30 ] )
+  do
+    sleep 2
+    echo -n "."
+    counter=$[ $counter + 1 ]
+    certificateDescription=$( aws acm --profile $profile --region $certificateRegion describe-certificate \
+                                  --certificate-arn $certificateArn )
+    certificateStatus=$( echo "$certificateDescription" | jq '.Certificate.Status' | tr -d '"' )
+  done
+  echo ""
+  echo "Certificate status = $certificateStatus"
```

Comment:
Forse sarebbe meglio uscire con errore in caso di mancata validazione (metti che sia incluso in uno task più grande)

Reply:
In alcuni casi le validazioni richiedono svariati minuti. 
I tempi di attesa potrebbero diventare inutilmente lunghi (a meno di non lanciare tutte le creazioni in parallelo)